cloud/kyup/kyup.py

```diff
@@ -0,0 +1,357 @@
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+DOCUMENTATION = '''
+---
+module: kyup
+short_description: create, terminate, start or stop a container in kyup.com, return container_id.
+description:
+    - Creates or terminates Kyup container. When created optionally waits for it to be 'running'. This module has a dependency on python-json and python-requests.
+version_added: "2.3"
+author: Marian Marinov <mm@kyup.com>
+options:
+  api_key:
+    description:
+      - Kyup.com API key. The key can also be supplied by setting the Environment variable KYUP_API_KEY.
+    required: true
+  enc_key:
+    description:
+      - Kyup.com encryption key. The key can also be supplied by setting the Environment variable KYUP_ENC_KEY.
+    required: false
+  action:
+    description:
+      - Action you want to perform on the container.
+    required: true
+  name:
+    description:
+      - Name of the container that you will create/operate on
+    required: true
+  ssh_keys:
+    description:
+      - Comma separated list of ssh key names, that have access to this container.
+    required: false
+  password:
+    description:
+      - Password for the root user of the container. Required when I(action) = 'create'.
+    required: false
+  image:
+    description:
+      - Template image to use when creating the container. Required when I(action) = 'create'.
+    required: false
+  dc_id:
+    description:
+      - DataCenter location ID. Required when I(action) = 'create'.
+    required: false
+    default: 1
+  mem:
+    description:
+      - Memory of the container in GB. Minimum 1 GB.
+    required: false
+    default: 1
+  hdd:
+    description:
+      - Storage size in GB. Minimum 20 GB.
+    required: false
+    default: 20
+  cpu_cores:
+    description:
+      - Number of CPU cores to be used. Minimum 2 CPU cores.
+    required: false
+    default: 2
+  bw:
+    description:
+      - Bandwidth limit for this container, in TB. Minimum 2 TB.
+    required: false
+    default: 2
+  storage_type:
+    description:
+      - Choose the storage type for this container.
+    required: false
+    default: local
+'''
+
+EXAMPLES = '''
+# Basic provisioning example
+---
+- hosts: kyup
+  tasks:
+  - name: ensure kyup container
+    kyup: >
+      api_key=XXXX
+      enc_key=YYYY
+      action=create
+      image="Centos Plain"
+      dc_id=1
+      name=avalon
+      password=e4a17265b28f6a9e8657
+      ssh_keys=hackman,piele
```

Comment:
As mentioned in a later comment, a `list` type is preferred for this parameter since it is actually a list of things.